src/models/state/wallet/wallet_state.rs

```diff
@@ -317,15 +345,18 @@ impl WalletState {
 
                 let spent_utxos = self.scan_for_spent_utxos(&tx.kernel).await;
 
+                let announced_from_addition = self
+                    .scan_addition_records_for_expected_utxos(&tx.kernel.outputs)
+                    .await;
```

Comment:
This variable name is confusing. This iterator yields `AnnouncedUtxo`s that are a) expected, and b) present in the transaction's list of outputs. In contrast, the next iterator yields `AnnouncedUtxo`s that are a) announced in the transaction's public announcements, and b) present in the transaction's list of outputs.

Consider binding both iterators (not just one) to variables named according to what sets them apart, *i.e.*, point (a).

As a more general commentary (not necessarily suggesting follow-up work): I think the root of the confusion might be the name of the struct `AnnouncedUtxo`. It certainly contains the right information. The name suggests that it can only originate from a public announcement. Expected UTXOs can also be an origin, however.

Reply:
AnnouncedUtxo naming predates this PR and is outside its scope, so will leave it be for now.

I will review the other suggestions, thx.

Reply:
renamed and commented both iterators for clarity.   code is now:

```

                // scan tx for utxo we can claim because we are expecting them (offchain)
                let announced_utxos_from_expected_utxos = self
                    .scan_addition_records_for_expected_utxos(&tx.kernel.outputs)
                    .await;

                // scan tx for utxo with public-announcements we can claim
                let announced_utxos_from_public_announcements =
                    self.scan_for_announced_utxos(&tx.kernel);

                let announced_utxos = announced_utxos_from_public_announcements
                    .chain(announced_utxos_from_expected_utxos)
                    .collect_vec();

```